src/openfermion/ops/_interaction_tensor.py

```diff
@@ -84,103 +85,114 @@ def two_body_basis_change(two_body_tensor, rotation_matrix):
-    terms is associated with the two-body Hamiltonian and 2-RDM and its
-    information is stored in two_body_tensor. Much of the functionality of this
-    class is redudant with FermionOperator but enables much more efficient
-    numerical computations in many cases, such as basis rotations.
+    """Class for storing tensor representations of operators that correspond
+    with multilinear polynomials in the fermionic ladder operators.
+    For instance, in a quadratic Hamiltonian (degree 2 polynomial),
+    there are only terms of the form a^\dagger_p a_q, and the coefficients
+    can be stored in an n_qubits x n_qubits matrix. Higher order terms would
+    be described with tensors of higher dimension. Note that each tensor must
+    have an even number of dimensions, since parity is conserved.
+    Much of the functionality of this class is redudant with FermionOperator
+    but enables much more efficient numerical computations in many cases,
+    such as basis rotations.
 
     Attributes:
-        n_qubits: The number of qubits on which the tensor
+        n_qubits(int): The number of qubits on which the tensor
             acts.
-        constant: A constant term in the operator given as a float.
-            For instance, the nuclear repulsion energy.
-        one_body_tensor: The coefficients of the 2D matrix terms. This is an
-            n_qubits x n_qubits numpy array of floats. For instance, the one
-            body term of MolecularOperator.
-        two_body_tensor: The coefficients of the 4D matrix terms. This is an
-            n_qubits x n_qubits x n_qubits x n_qubits numpy array offloats.
-            For instance, the two body term of MolecularOperator.
+        constant(complex): A constant term in the operator given as a complex
+            number. For instance, the nuclear repulsion energy.
+        n_body_tensors(dict): A dictionary storing the tensors describing
+            n-body interactions. For instance, n_body_tensors[2] is a
+            n_qubits x n_qubits x n_qubits x n_qubits numpy array of complex
+            numbers.
     """
 
-    def __init__(self, constant, one_body_tensor, two_body_tensor):
+    def __init__(self, constant, *args, n_body_tensors=None, n_qubits=None):
         """Initialize the InteractionTensor class.
 
         Args:
-            constant: A constant term in the operator given as a
-                float. For instance, the nuclear repulsion energy.
-            one_body_tensor: The coefficients of the 2D matrix terms. This
-                is an n_qubits x n_qubits numpy array of floats. For
-                instance, the one body term of MolecularOperator.
-            two_body_tensor: The coefficients of the 4D matrix terms. This is
-                an n_qubits x n_qubits x n_qubits x n_qubits numpy array of
-                floats. For instance, the two body term of MolecularOperator.
+            constant(complex): A constant term in the operator given as a
```

Comment:
It represents the coefficient of the identity operator.

In the context of Hamiltonians, the constant represents an energy shift. I.e., it simply shifts the eigenvalues of the Hamiltonian, without affecting the eigenvectors. In that case it would just be a real number. I couldn't decide whether to insist that it's a float or let it be complex.

Reply:
👍